Source/WebKit/UIProcess/RemoteLayerTree/mac/RemoteLayerTreeDrawingAreaProxyMac.mm

```diff
@@ -128,15 +128,45 @@ explicit RemoteLayerTreeDisplayLinkClient(WebPageProxyIdentifier pageID)
     observerID = { };
 }
 
+void RemoteLayerTreeDrawingAreaProxyMac::layoutBannerLayers()
+{
+    float totalContentHeight = size().height();
```

Comment:
This isn't the right way to get the total height. `size()` on the DrawingAreaProxy is the size of the WKWebView, not the size of the scrolled contents. This bug should be visible if you add banners while on a long scrolling page.

The right height is probably `RemoteLayerTreeTransaction::contentsSize()` but you should figure out if this already factors in banners.

Reply:
Yes this is right. `transaction.contentsSize().height()` is used instead. I tested the log scrolling page case and it worked as expected.